apiserver/apiserver.go

```diff
@@ -1141,7 +1141,20 @@ func (srv *Server) serveConn(
 	st, err := statePool.Get(resolvedModelUUID)
 	if err == nil {
 		defer st.Release()
-		handler, err = newAPIHandler(srv, st.State, conn, serviceFactory, srv.shared.serviceFactoryGetter, tracer, objectStore, controllerObjectStore, modelUUID, connectionID, host)
+		handler, err = newAPIHandler(
+			srv,
+			st.State,
+			conn,
+			serviceFactory,
```

Comment:
I loath the fact that we're handing around both a concrete service factory and a getter, as well as 2 object stores plus a getter.

Can't we at least push the concrete acquisitions inside the constructor here to make fewer arguments?

Reply:
I honestly this we should just feel the pain here. This does need refactoring, but I'm not in any hurry to do it until we've got a solid plan for it.